File: src/types/novel.ts
export interface Novel {
  id: number
  title: string
  author: string
  category: string
  rawText: string
  cleanedText: string
  isFavorite: boolean
  /** 'zh' = Chinese novel (match by Chinese definition) |
   *  'en' = English novel (match by English word) */
  language: 'zh' | 'en'
  createdAt: string
  updatedAt: string
}

export interface NovelFormData {
  title: string
  author: string
  category: string
  rawText?: string
  cleanedText?: string
  language?: 'zh' | 'en'
}

export interface Chapter {
  id: number
  novelId: number
  title: string
  content: string
  sortOrder: number
  startIndex: number
  createdAt: string
}

export interface ImportResult {
  chapters: Chapter[]
  rawText: string
  cleanedText: string
  detectedTitle: string
}

File: src-tauri/src/models/novel.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Novel {
    pub id: i64,
    pub title: String,
    pub author: String,
    pub category: String,
    pub raw_text: String,
    pub cleaned_text: String,
    pub is_favorite: bool,
    /// 'zh' = Chinese novel (match vocab by Chinese definition) |
    /// 'en' = English novel (match vocab by English word)
    pub language: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Chapter {
    pub id: i64,
    pub novel_id: i64,
    pub title: String,
    pub content: String,
    pub sort_order: i32,
    pub start_index: usize,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ImportResult {
    pub chapters: Vec<Chapter>,
    pub raw_text: String,
    pub cleaned_text: String,
    pub detected_title: String,
}
